src/compass/s1_geo_stack.py

```diff
@@ -0,0 +1,468 @@
+        Path to DEM to use for processing
+    work_dir: str
+        Path to working directory for temp and final results
+    flatten: bool
+        Flag to enable/disable flattening
+    enable_rss: bool
+        Flag to enable range split-spectrum
+    low-band: float
+        Low sub-image bandwidth (in Hz) for split-spectrum
+    high-band: float
+        High sub-image bandwidth (in Hz) for split-spectrum
+    pol: str
+        Polarizations to process: co-pol, dual-pol, cross-pol
+    x_spac: float
+        Spacing of geocoded burst along X-direction
+    y_spac: float
+        Spacing of geocoded burst along Y-direction
+    '''
+    # Load default runconfig and fill it with user-defined options
+    yaml_path = f'{helpers.WORKFLOW_SCRIPTS_DIR}/defaults/s1_cslc_geo.yaml'
+    with open(yaml_path, 'r') as stream:
+        yaml_cfg = yaml.safe_load(stream)
+
+    groups = yaml_cfg['runconfig']['groups']
+    inputs = groups['input_file_group']
+    product = groups['product_path_group']
+    process = groups['processing']
+    geocode = process['geocoding']
+    rss = process['range_split_spectrum']
+
+    # Allocate Inputs
+    burst = burst_map_row.burst
+    inputs['safe_file_path'] = [burst_map_row.zip_file]
+    inputs['orbit_file_path'] = [burst_map_row.orbit_path]
+    inputs['burst_id'] = burst.burst_id
+    groups['dynamic_ancillary_file_group']['dem_file'] = dem_file
+
+    # Product path
+    product['product_path'] = work_dir
+    product['scratch_path'] = f'{work_dir}/scratch'
+    product['sas_output_file'] = work_dir
+
+    # Geocoding
+    process['polarization'] = pol
+    geocode['flatten'] = flatten
+    geocode['x_posting'] = x_spac
+    geocode['y_posting'] = y_spac
+
+    geocode['top_left']['x'] = burst_map_row.x_top_left
+    geocode['top_left']['y'] = burst_map_row.y_top_left
+    geocode['bottom_right']['x'] = burst_map_row.x_bottom_right
+    geocode['bottom_right']['y'] = burst_map_row.y_bottom_right
+    # geocode['x_snap'] = None
+    # geocode['y_snap'] = None
+    geocode['output_epsg'] = burst_map_row.epsg
+
+    # Range split spectrum
+    rss['enabled'] = enable_rss
+    rss['low_band_bandwidth'] = low_band
+    rss['high_band_bandwidth'] = high_band
+
+    date_str = burst.sensing_start.strftime("%Y%m%d")
+    os.makedirs(f'{work_dir}/runconfigs', exist_ok=True)
+    runconfig_path = f'{work_dir}/runconfigs/geo_runconfig_{date_str}_{burst.burst_id}.yaml'
+    with open(runconfig_path, 'w') as yaml_file:
+        yaml.dump(yaml_cfg, yaml_file, default_flow_style=False)
+    return runconfig_path
+
+
+def _filter_by_date(zip_file_list, start_date, end_date, exclude_dates):
+    '''
+    Filter list of zip files based on date
+    Parameters:
+    ----------
+    zip_file_list: list
+        List of zip files to filter
+    start_date: str
+        Start date in YYYYMMDD format
+    end_date: str
+        End date in YYYYMMDD format
+    exclude_dates: list
+        List of dates to exclude
```

Comment:
```suggestion

    Parameters:
    ----------
    zip_file_list: list
        List of zip files to filter
    start_date: str
        Start date in YYYYMMDD format
    end_date: str
        End date in YYYYMMDD format
    exclude_dates: list
        List of dates to exclude
        
```
Extra lines for readability.